odoo/addons/base/tests/test_ir_mail_server_smtpd.py

```diff
@@ -0,0 +1,651 @@
+import logging
+import shutil
+import smtplib
+import ssl
+import subprocess
+import tempfile
+import unittest
+import warnings
+from base64 import b64encode
+from collections import namedtuple
+from pathlib import Path
+from socket import getaddrinfo
+from unittest.mock import patch
+
+from odoo.exceptions import UserError
+from odoo.tools import file_path, mute_logger
+from .common import TransactionCaseWithUserDemo
+
+try:
+    import aiosmtpd
+    import aiosmtpd.controller
+    import aiosmtpd.smtp
+    import aiosmtpd.handlers
+except ImportError:
+    aiosmtpd = None
+
+
+PASSWORD = 'secretpassword'
+PORT = 8025
```

Comment:
can you have many instances of the test running in // ?
(if yes the port can be random, otherwise we can keep it like this)

Reply:
The tests only take 8 seconds to run, the underlying smtpd threads spawn quite fast actually. I'm not sure we'll benefit from running them in //, also considering that the logs will be mixed up, and that I'll need to add some code to "retry binding another port" in case of `OSError: address already in use`. Or I could do `PORT = 8025 + version_info[0] * 10 + version_info[1]` :joy:

Reply:
> I'm not sure we'll benefit from running them in //

Ah yes, I didn't want to run them in //, just if those tests are run in // on the runbot, it might fail because of the port being already used

But I guess it shouldn't happen

Reply:
they don't run in // ;)